bbc/simorgh
# WSTEAMA-608: Migrate Optimo MAP, FIX and IDX to use BFF ATI data

Resolves JIRA #WSTEAMA-608

Overall changes
======
_A very high-level summary of easily-reproducible changes that can be understood by non-devs, and why these changes where made._

Code changes
======

- Migrate FIX, IDX and Optimo Media Article Page to use ATI data from the BFF.
- Remove redundant Optimo articles ATI params module and accompanying tests
    - `src/app/components/ATIAnalytics/params/article/buildParams.ts`
    - `src/app/components/ATIAnalytics/params/article/buildParams.test.ts`
- Update pertinent tests and fixtures.

#### Note:
- `buildIndexPageATIUrl` cannot be removed at this time since the [FRONT_PAGE](https://github.com/bbc/simorgh/blob/6c30b0648fa7ab66d17557e44339e4ad35d3745b/src/app/components/ATIAnalytics/params/index.ts#L76) uses it.

Testing
======
1. Visit the pages linked below.
    ```
    Optimo Media Article Page: http://localhost:7080/pidgin/articles/cw0x29n2pvqo
    IDX: http://localhost:7080/persian/afghanistan
    IDX: http://localhost:7080/ukrainian/ukraine_in_russian
    FIX: http://localhost:7080/urdu/science-51314202
    ```
2. Check that the following beacons are triggered .

| Page Type  | Beacons |
| ----------- | --------- |
| Optimo Media Article Page | Views: <ul><li>page views.</li><li>latest.</li><li>topics.</li><li>related content.</li></ul> <br> Click: <ul><li>latest.</li><li>topics.</li><li>related content.</li></ul> |
| IDX | Views: <ul><li>page views.</li><li>radio-schedule.</li><li>most-read.</li></ul> <br> Click: <ul><li>radio-schedule.</li><li>most-read.</li></ul> |
| FIX | Views: <ul><li>page views.</li></ul>|

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
